mmtf-codec/src/test/java/org/rcsb/mmtf/encoder/TestReducedEncoderComprehensive.java

```diff
@@ -0,0 +1,163 @@
+package org.rcsb.mmtf.encoder;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.text.ParseException;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Test;
+import org.rcsb.mmtf.api.StructureDataInterface;
+import org.rcsb.mmtf.codec.Utils;
+import org.rcsb.mmtf.decoder.GenericDecoder;
+import org.rcsb.mmtf.decoder.ReaderUtils;
+
+public class TestReducedEncoderComprehensive {
+
+	@Test
+	public void test() throws IOException, ParseException {
+		StructureDataInterface full = getDefaultFullData();
+		StructureDataInterface reduced = ReducedEncoder.getReduced(getDefaultFullData());
+		compareFullVsReduced(full.getStructureId(), full, reduced);
```

Comment:
How about including a few more cases here, not only 4cup. E.g. structure with nucleotides, structure with water marked as non-polymer, structure with alt conformations, etc...